app/client/src/components/shared/MapWidgets.tsx

```diff
@@ -1042,11 +1045,25 @@ function MapWidgets({
     visibleLayers,
   ]);
 
+  // Focus on the add/save data widget when it is first opened
   useEffect(() => {
     if (!addSaveDataWidgetVisible) return;
     document.getElementById('add-save-data-widget')?.focus();
   }, [addSaveDataWidgetVisible]);
 
+  // Reset the add/save data widget values when users change pages
+  useEffect(() => {
+    setActiveTabIndex(0);
+    setAddSaveDataWidgetVisible(false);
+    setSaveAsName('');
+    setSaveDescription('');
+  }, [
+    setActiveTabIndex,
+    setAddSaveDataWidgetVisible,
+    setSaveAsName,
+    setSaveDescription,
+  ]);
+
```

Comment:
This does the trick! As an alternative, we could also drop the `AddSaveDataWidgetProvider` down to the page level, rather than wrapping all the routes with it (this could probably be done to several other providers, too). This would adhere to the "keep state as low as possible" principle, and it would save on maintenance if other variables are added to the state down the road. The added layers would be cleared, though.